### DISCUSSION TITLE
vscode to integrate jupyter

### DISCUSSION THREAD

**Discussion by Pythoneryeah** [coder/code-server#6337] [Q&A] [answered]
Hello everyone, when I use vscode to integrate jupyter, click the red box button in the screenshot, why does the code editing window not pop up?

![image](https://github.com/coder/code-server/assets/37642828/e7cff7e9-d199-41e8-a21f-c5e6e7ec1a0e)

**Comment by code-asher** [comment]
It works for me (code-server v4.14.1 using Firefox and Linux).  Do 
you get any errors in the logs or browser console?

**Reply by Pythoneryeah** [reply]
Looks like a display bug? My jupyter opens this file to see the created cell


![image](https://github.com/coder/code-server/assets/37642828/23c71821-5111-490f-90f7-fc205bab17a7)

**Reply by Pythoneryeah** [reply]
![image](https://github.com/coder/code-server/assets/37642828/51e1e838-5f11-4081-a0e4-514eabb6c3c2)

this is vscode and jupyter  logs.

**Reply by code-asher** [reply]
Ahh interesting, so it does create the cell but is not showing them to you.

Could you also post the logs from the browser console when you click the button?  (You can get to it from the browser's developer tools.)

**Comment by benz0li** [comment, marked as answer]
Are you using HTTPS? This is required for Jupyter notebooks.  
ℹ️ Fully qualified domain name (FQDN) required, too. TLS for an IP is not sufficient.

HTTP only works for `127.0.0.1` / `localhost.`

Cross reference: https://github.com/coder/code-server/issues/5475#issuecomment-1229529712

**Reply by Pythoneryeah** [reply]
Thank you. Indeed, when I access the local service, it appears fine. The problem is only on the remote server. What do I need to do if I want to access the remote code-server integration jupyter?

**Reply by benz0li** [reply]
You need a [TLS termination proxy](https://en.wikipedia.org/wiki/TLS_termination_proxy) in front of code-server. Common web servers to achieve this:

- Nginx: https://docs.nginx.com/nginx/admin-guide/security-controls/terminating-ssl-http/
- Apache: https://httpd.apache.org/docs/trunk/ssl/ssl_howto.html
- Træfik: https://doc.traefik.io/traefik/https/tls/
- HAProxy: https://www.haproxy.com/blog/haproxy-ssl-termination

**Reply by Pythoneryeah** [reply]
Thank you, I solved the problem by referring to your solution
